### PR TITLE
Add enhancements to find, edit features and add parse ics feature

### PR DESCRIPTION
**Pull request by mathanmahe**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -290,6 +290,81 @@ Expected outcome:
     Noted I have removed this module from your schedule:
     [Module] CS2113 Lecture: Thursday, 2pm-4pm
 
+#### Editing a module: `edit`
```
Comment (OzairHasan) [inline docs/UserGuide.md:293]: Very clear examples of usage in the UG.

**Review by OzairHasan — COMMENTED** [review]
LGTM

File: src/main/java/seedu/allonus/modules/ModuleCalendarReader.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    String[] summary;
+    StringBuilder timeSlot;
+    Date dateStart = null;
+    Date dateEnd;
+    CalendarBuilder builder;
+    SimpleDateFormat dateFormat;
+    Calendar calendar;
+    TimeZone sgTimeZone = TimeZone.getTimeZone(SINGAPORE_TIMEZONE_KEYWORD);
+
+    public ArrayList<Module> parseIcsCalendar() {
```
Comment (wli-linda) [inline src/main/java/seedu/allonus/modules/ModuleCalendarReader.java:86]: Is it possible to extract some methods out and make this method shorter?

File: src/main/java/seedu/allonus/modules/ModuleCalendarReader.java [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        timeSlot.append(new SimpleDateFormat(TIME_FORMAT_WITH_AMPM).format(dateStart));
+
+        java.util.Calendar calendarNew = java.util.Calendar.getInstance();
+        calendarNew.setTime(dateStart);
+        int day = calendarNew.get(java.util.Calendar.DAY_OF_WEEK);
+        moduleDay = convertDayToString(day);
+    }
+
+    private void getModuleSummary(Property property) {
+        summary = property.getValue().split(" ",2);
```
Comment (wli-linda) [inline src/main/java/seedu/allonus/modules/ModuleCalendarReader.java:183]: I think you might be missing a space here for `.split(" ", 2)`.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -290,6 +290,81 @@ Expected outcome:
...
+
+`done`
+
+| Parameter      | Description                       |
+|----------------|-----------------------------------|
+| ```<index> ``` | Index of the module to be edited |
+| ```<MODULE_CODE> ``` | New code for the module                  |
+| ```<CATEGORY> ```    | New category of the module               |
+| ```<DAY> ```         | New class day of the week for the module |
+| ```<TIME> ```        | New class timing for the module          |
```
Comment (YitHien) [inline docs/UserGuide.md:314]: Nice tabling format for User Guide! I'll add this to my section as well

File: src/main/java/seedu/allonus/modules/ModuleCalendarReader.java [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+            default:
+                throw new ModuleDayException(INCORRECT_DAY_ERROR_MESSAGE);
+            }
+        }  catch (ModuleDayException e) {
+            System.out.println(e.getMessage());
+        }
+        return day;
+    }
+
+}
```
Comment (YitHien) [inline src/main/java/seedu/allonus/modules/ModuleCalendarReader.java:224]: Great work on implementing the calendar reader!

**Review by YitHien — COMMENTED** [review]
Great feature update on the calendar parsing! LGTM

File: src/main/java/seedu/allonus/modules/StudyManager.java [hunk 6]
```diff
@@ -226,35 +254,43 @@ private void editModuleRunner(TextUi ui, int moduleIndex) {
...
 
+    private void printEditWelcomeMessage(Module moduleToEdit) {
+        printMessage(EDIT_MODULE_OPENING_MESSAGE);
+        printMessage(moduleToEdit.toString());
+        printMessage(EDIT_MODULE_CHOOSE_MESSAGE);
+    }
+
     private void editModuleTime(Module moduleToEdit, String editUserInput) {
-        String moduleTime = editUserInput.replace(MODULE_TIME_DELIMITER,"");
+        String moduleTime = editUserInput.replace(MODULE_TIME_DELIMITER,EMPTY_STRING);
```
Comment (wli-linda) [inline src/main/java/seedu/allonus/modules/StudyManager.java:275]: Also might need a space here between the two arguments in `.replace()`

File: src/main/java/seedu/allonus/modules/StudyManager.java [hunk 7]
```diff
@@ -324,19 +360,54 @@ public Module addModuleParser(String userInput) {
...
+     * Finds a module in the list using a find query.
+     * @param userInput String containing the find command and find query.
+     */
     public void findModule(String userInput) {
-        String moduleKeyword = userInput.replace(FIND_COMMAND + " ","");
+        try {
+            String moduleKeyword = validateFindQuery(userInput);
+            moduleKeyword = moduleKeyword.toLowerCase();
+            ArrayList<Module> matches = getFindMatches(moduleKeyword);
+            if (matches.size() == 0) {
```
Comment (wli-linda) [inline src/main/java/seedu/allonus/modules/StudyManager.java:372]: Can consider using `.isEmpty` instead

File: src/main/java/seedu/allonus/modules/StudyManager.java [hunk 8]
```diff
@@ -324,19 +360,54 @@ public Module addModuleParser(String userInput) {
...
+     * @return valid module keyword to be used as search query.
+     * @throws InvalidFindInputException for empty spaces and special characters in find query.
+     */
+    private String validateFindQuery(String userInput) throws InvalidFindInputException {
+        String moduleKeyword = userInput.replace(FIND_COMMAND + STRING_SPACE_CHARACTER, EMPTY_STRING);
+        if (moduleKeyword.equals(STRING_SPACE_CHARACTER) || moduleKeyword.equals(EMPTY_STRING)
+                || !userInput.contains(" ")) {
+            throw new InvalidFindInputException("You have not entered a search keyword to find modules!");
+        } else if (moduleKeyword.equals(":") || moduleKeyword.equals("[") || moduleKeyword.equals("]")
+                || moduleKeyword.equals(",") || moduleKeyword.equals("-")) {
```
Comment (wli-linda) [inline src/main/java/seedu/allonus/modules/StudyManager.java:406]: Can consider extracting this condition out to make it more understandable!

**Review by wli-linda — COMMENTED** [review]
LGTM, just a few non-urgent code quality improvements!

**Pull request merged**